Build error on SIP Version 4.19.23

I am trying to build python_orocos_kdl for ROS Noetic. It wont build on Arch with SIP Version 4.19.23.

Its also reported [here](https://github.com/ros-noetic-arch/ros-noetic-python-orocos-kdl/issues/1)

```
==> Making package: ros-noetic-python-orocos-kdl 1.4.0-3 (Fr 12 Jun 2020 21:31:30 CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found ros-noetic-python-orocos-kdl-1.4.0.tar.gz
==> Validating source files with sha256sums...
    ros-noetic-python-orocos-kdl-1.4.0.tar.gz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting ros-noetic-python-orocos-kdl-1.4.0.tar.gz with bsdtar
==> Sources are ready.
==> Making package: ros-noetic-python-orocos-kdl 1.4.0-3 (Fr 12 Jun 2020 21:31:31 CEST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Starting build()...
-- The C compiler identification is GNU 10.1.0
-- The CXX compiler identification is GNU 10.1.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PythonInterp: /usr/bin/python (found suitable version "3.8.3", minimum required is "2") 
-- Found PythonLibs: /usr/lib/libpython3.8.so (found suitable version "3.8.3", minimum required is "3.8") 
-- Found SIP version: 4.19.23
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CATKIN_BUILD_BINARY_PACKAGE
    SETUPTOOLS_DEB_LAYOUT


-- Build files have been written to: /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build
[ 10%] Generating PyKDL/sipPyKDLpart0.cpp, PyKDL/sipPyKDLpart1.cpp, PyKDL/sipPyKDLpart2.cpp, PyKDL/sipPyKDLpart3.cpp, PyKDL/sipPyKDLpart4.cpp, PyKDL/sipPyKDLpart5.cpp, PyKDL/sipPyKDLpart6.cpp, PyKDL/sipPyKDLpart7.cpp

sip: Deprecation warning: /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/PyKDL.sip:23: %Module version numbers are deprecated and ignored
Scanning dependencies of target python_module_PyKDL
[ 20%] Building CXX object CMakeFiles/python_module_PyKDL.dir/PyKDL/sipPyKDLpart0.o
[ 40%] Building CXX object CMakeFiles/python_module_PyKDL.dir/PyKDL/sipPyKDLpart2.o
[ 40%] Building CXX object CMakeFiles/python_module_PyKDL.dir/PyKDL/sipPyKDLpart1.o
[ 50%] Building CXX object CMakeFiles/python_module_PyKDL.dir/PyKDL/sipPyKDLpart3.o
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart1.cpp: In function ‘void assign_ChainJntToJacSolver(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart1.cpp:2124:119: error: use of deleted function ‘KDL::ChainJntToJacSolver& KDL::ChainJntToJacSolver::operator=(const KDL::ChainJntToJacSolver&)’
 2124 |         };
      |                                                                                                                       ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:567:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainjnttojacsolver.hpp:41:11: note: ‘KDL::ChainJntToJacSolver& KDL::ChainJntToJacSolver::operator=(const KDL::ChainJntToJacSolver&)’ is implicitly deleted because the default definition would be ill-formed:
   41 |     class ChainJntToJacSolver : public SolverI
      |           ^~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainjnttojacsolver.hpp:41:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainJntToJacSolver::chain’, cannot use default assignment operator
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart3.cpp: In function ‘void assign_ChainIkSolverPos_NR(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart3.cpp:204:119: error: use of deleted function ‘KDL::ChainIkSolverPos_NR& KDL::ChainIkSolverPos_NR::operator=(const KDL::ChainIkSolverPos_NR&)’
  204 |     reinterpret_cast< ::ChainIkSolverPos_NR *>(sipDst)[sipDstIdx] = *reinterpret_cast< ::ChainIkSolverPos_NR *>(sipSrc);
      |                                                                                                                       ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:386:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_nr.hpp:38:11: note: ‘KDL::ChainIkSolverPos_NR& KDL::ChainIkSolverPos_NR::operator=(const KDL::ChainIkSolverPos_NR&)’ is implicitly deleted because the default definition would be ill-formed:
   38 |     class ChainIkSolverPos_NR : public ChainIkSolverPos
      |           ^~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_nr.hpp:38:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainIkSolverPos_NR::chain’, cannot use default assignment operator
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_nr.hpp:38:11: error: non-static reference member ‘KDL::ChainIkSolverVel& KDL::ChainIkSolverPos_NR::iksolver’, cannot use default assignment operator
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_nr.hpp:38:11: error: non-static reference member ‘KDL::ChainFkSolverPos& KDL::ChainIkSolverPos_NR::fksolver’, cannot use default assignment operator
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart3.cpp: In function ‘void assign_ChainFkSolverVel_recursive(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart3.cpp:1176:133: error: use of deleted function ‘KDL::ChainFkSolverVel_recursive& KDL::ChainFkSolverVel_recursive::operator=(const KDL::ChainFkSolverVel_recursive&)’
 1176 | {
      |                                                                                                                                     ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:352:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainfksolvervel_recursive.hpp:37:11: note: ‘KDL::ChainFkSolverVel_recursive& KDL::ChainFkSolverVel_recursive::operator=(const KDL::ChainFkSolverVel_recursive&)’ is implicitly deleted because the default definition would be ill-formed:
   37 |     class ChainFkSolverVel_recursive : public ChainFkSolverVel
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainfksolvervel_recursive.hpp:37:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainFkSolverVel_recursive::chain’, cannot use default assignment operator
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart3.cpp: In function ‘void assign_ChainFkSolverPos_recursive(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart3.cpp:1492:133: error: use of deleted function ‘KDL::ChainFkSolverPos_recursive& KDL::ChainFkSolverPos_recursive::operator=(const KDL::ChainFkSolverPos_recursive&)’
 1492 |     if (sipIsOwnedByPython(sipSelf))
      |                                                                                                                                     ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:339:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainfksolverpos_recursive.hpp:36:11: note: ‘KDL::ChainFkSolverPos_recursive& KDL::ChainFkSolverPos_recursive::operator=(const KDL::ChainFkSolverPos_recursive&)’ is implicitly deleted because the default definition would be ill-formed:
   36 |     class ChainFkSolverPos_recursive : public ChainFkSolverPos
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainfksolverpos_recursive.hpp:36:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainFkSolverPos_recursive::chain’, cannot use default assignment operator
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp: In function ‘void assign_ChainIkSolverVel_pinv_nso(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp:418:131: error: use of deleted function ‘KDL::ChainIkSolverVel_pinv_nso& KDL::ChainIkSolverVel_pinv_nso::operator=(const KDL::ChainIkSolverVel_pinv_nso&)’
  418 |     reinterpret_cast< ::ChainIkSolverVel_pinv_nso *>(sipDst)[sipDstIdx] = *reinterpret_cast< ::ChainIkSolverVel_pinv_nso *>(sipSrc);
      |                                                                                                                                   ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:529:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv_nso.hpp:46:11: note: ‘KDL::ChainIkSolverVel_pinv_nso& KDL::ChainIkSolverVel_pinv_nso::operator=(const KDL::ChainIkSolverVel_pinv_nso&)’ is implicitly deleted because the default definition would be ill-formed:
   46 |     class ChainIkSolverVel_pinv_nso : public ChainIkSolverVel
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv_nso.hpp:46:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainIkSolverVel_pinv_nso::chain’, cannot use default assignment operator
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv_nso.hpp:46:11: error: use of deleted function ‘KDL::ChainJntToJacSolver& KDL::ChainJntToJacSolver::operator=(const KDL::ChainJntToJacSolver&)’
In file included from /opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv_nso.hpp:26,
                 from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:529:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainjnttojacsolver.hpp:41:11: note: ‘KDL::ChainJntToJacSolver& KDL::ChainJntToJacSolver::operator=(const KDL::ChainJntToJacSolver&)’ is implicitly deleted because the default definition would be ill-formed:
   41 |     class ChainJntToJacSolver : public SolverI
      |           ^~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainjnttojacsolver.hpp:41:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainJntToJacSolver::chain’, cannot use default assignment operator
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart1.cpp: In function ‘void assign_ChainIkSolverVel_pinv_givens(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart1.cpp:2440:137: error: use of deleted function ‘KDL::ChainIkSolverVel_pinv_givens& KDL::ChainIkSolverVel_pinv_givens::operator=(const KDL::ChainIkSolverVel_pinv_givens&)’
 2440 |             sipCpp = new sipChainIkSolverVel_pinv_givens(*a0);
      |                                                                                                                                         ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:554:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv_givens.hpp:24:11: note: ‘KDL::ChainIkSolverVel_pinv_givens& KDL::ChainIkSolverVel_pinv_givens::operator=(const KDL::ChainIkSolverVel_pinv_givens&)’ is implicitly deleted because the default definition would be ill-formed:
   24 |     class ChainIkSolverVel_pinv_givens : public ChainIkSolverVel
      |           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv_givens.hpp:24:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainIkSolverVel_pinv_givens::chain’, cannot use default assignment operator
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv_givens.hpp:24:11: error: use of deleted function ‘KDL::ChainJntToJacSolver& KDL::ChainJntToJacSolver::operator=(const KDL::ChainJntToJacSolver&)’
make[2]: *** [CMakeFiles/python_module_PyKDL.dir/build.make:149: CMakeFiles/python_module_PyKDL.dir/PyKDL/sipPyKDLpart3.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [CMakeFiles/python_module_PyKDL.dir/build.make:123: CMakeFiles/python_module_PyKDL.dir/PyKDL/sipPyKDLpart1.o] Error 1
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp: In function ‘void assign_ChainIkSolverPos_LMA(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp:752:121: error: use of deleted function ‘KDL::ChainIkSolverPos_LMA& KDL::ChainIkSolverPos_LMA::operator=(const KDL::ChainIkSolverPos_LMA&)’
  752 | 
      |                                                                                                                         ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:515:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_lma.hpp:64:7: note: ‘KDL::ChainIkSolverPos_LMA& KDL::ChainIkSolverPos_LMA::operator=(const KDL::ChainIkSolverPos_LMA&)’ is implicitly deleted because the default definition would be ill-formed:
   64 | class ChainIkSolverPos_LMA : public KDL::ChainIkSolverPos
      |       ^~~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_lma.hpp:64:7: error: non-static reference member ‘const KDL::Chain& KDL::ChainIkSolverPos_LMA::chain’, cannot use default assignment operator
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp: In function ‘void assign_ChainIkSolverVel_wdls(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp:1234:123: error: use of deleted function ‘KDL::ChainIkSolverVel_wdls& KDL::ChainIkSolverVel_wdls::operator=(const KDL::ChainIkSolverVel_wdls&)’
 1234 |     {
      |                                                                                                                           ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:428:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_wdls.hpp:63:11: note: ‘KDL::ChainIkSolverVel_wdls& KDL::ChainIkSolverVel_wdls::operator=(const KDL::ChainIkSolverVel_wdls&)’ is implicitly deleted because the default definition would be ill-formed:
   63 |     class ChainIkSolverVel_wdls : public ChainIkSolverVel
      |           ^~~~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_wdls.hpp:63:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainIkSolverVel_wdls::chain’, cannot use default assignment operator
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_wdls.hpp:63:11: error: use of deleted function ‘KDL::ChainJntToJacSolver& KDL::ChainJntToJacSolver::operator=(const KDL::ChainJntToJacSolver&)’
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp: In function ‘void assign_ChainIkSolverVel_pinv(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp:1556:123: error: use of deleted function ‘KDL::ChainIkSolverVel_pinv& KDL::ChainIkSolverVel_pinv::operator=(const KDL::ChainIkSolverVel_pinv&)’
 1556 | }
      |                                                                                                                           ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:415:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv.hpp:40:11: note: ‘KDL::ChainIkSolverVel_pinv& KDL::ChainIkSolverVel_pinv::operator=(const KDL::ChainIkSolverVel_pinv&)’ is implicitly deleted because the default definition would be ill-formed:
   40 |     class ChainIkSolverVel_pinv : public ChainIkSolverVel
      |           ^~~~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv.hpp:40:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainIkSolverVel_pinv::chain’, cannot use default assignment operator
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolvervel_pinv.hpp:40:11: error: use of deleted function ‘KDL::ChainJntToJacSolver& KDL::ChainJntToJacSolver::operator=(const KDL::ChainJntToJacSolver&)’
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp: In function ‘void assign_ChainIkSolverPos_NR_JL(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart2.cpp:1884:125: error: use of deleted function ‘KDL::ChainIkSolverPos_NR_JL& KDL::ChainIkSolverPos_NR_JL::operator=(const KDL::ChainIkSolverPos_NR_JL&)’
 1884 | static void *init_type_ChainIkSolverPos_NR_JL(sipSimpleWrapper *sipSelf, PyObject *sipArgs, PyObject *sipKwds, PyObject **sipUnused, PyObject **, PyObject **sipParseErr)
      |                                                                                                                             ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/kinfam.sip:400:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_nr_jl.hpp:40:11: note: ‘KDL::ChainIkSolverPos_NR_JL& KDL::ChainIkSolverPos_NR_JL::operator=(const KDL::ChainIkSolverPos_NR_JL&)’ is implicitly deleted because the default definition would be ill-formed:
   40 |     class ChainIkSolverPos_NR_JL : public ChainIkSolverPos
      |           ^~~~~~~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_nr_jl.hpp:40:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainIkSolverPos_NR_JL::chain’, cannot use default assignment operator
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_nr_jl.hpp:40:11: error: non-static reference member ‘KDL::ChainIkSolverVel& KDL::ChainIkSolverPos_NR_JL::iksolver’, cannot use default assignment operator
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainiksolverpos_nr_jl.hpp:40:11: error: non-static reference member ‘KDL::ChainFkSolverPos& KDL::ChainIkSolverPos_NR_JL::fksolver’, cannot use default assignment operator
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart0.cpp: In function ‘void assign_ChainDynParam(void*, Py_ssize_t, void*)’:
/home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/build/PyKDL/sipPyKDLpart0.cpp:2649:107: error: use of deleted function ‘KDL::ChainDynParam& KDL::ChainDynParam::operator=(const KDL::ChainDynParam&)’
 2649 |         sipNameNr_ChainDynParam,
      |                                                                                                           ^
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/dynamics.sip:65:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chaindynparam.hpp:47:11: note: ‘KDL::ChainDynParam& KDL::ChainDynParam::operator=(const KDL::ChainDynParam&)’ is implicitly deleted because the default definition would be ill-formed:
   47 |     class ChainDynParam : SolverI
      |           ^~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chaindynparam.hpp:47:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainDynParam::chain’, cannot use default assignment operator
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chaindynparam.hpp:47:11: error: use of deleted function ‘KDL::ChainIdSolver_RNE& KDL::ChainIdSolver_RNE::operator=(const KDL::ChainIdSolver_RNE&)’
In file included from /opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chaindynparam.hpp:25,
                 from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/dynamics.sip:65:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainidsolver_recursive_newton_euler.hpp:40:11: note: ‘KDL::ChainIdSolver_RNE& KDL::ChainIdSolver_RNE::operator=(const KDL::ChainIdSolver_RNE&)’ is implicitly deleted because the default definition would be ill-formed:
   40 |     class ChainIdSolver_RNE : public ChainIdSolver{
      |           ^~~~~~~~~~~~~~~~~
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chainidsolver_recursive_newton_euler.hpp:40:11: error: non-static reference member ‘const KDL::Chain& KDL::ChainIdSolver_RNE::chain’, cannot use default assignment operator
In file included from /home/benjamin/.cache/yay/ros-noetic-python-orocos-kdl/src/orocos_kinematics_dynamics-1.4.0/python_orocos_kdl/PyKDL/dynamics.sip:65:
/opt/ros/noetic/share/orocos_kdl/cmake/../../../include/kdl/chaindynparam.hpp:47:11: error: use of deleted function ‘KDL::ChainIdSolver_RNE& KDL::ChainIdSolver_RNE::operator=(const KDL::ChainIdSolver_RNE&)’
   47 |     class ChainDynParam : SolverI
      |           ^~~~~~~~~~~~~
make[2]: *** [CMakeFiles/python_module_PyKDL.dir/build.make:136: CMakeFiles/python_module_PyKDL.dir/PyKDL/sipPyKDLpart2.o] Error 1
make[2]: *** [CMakeFiles/python_module_PyKDL.dir/build.make:110: CMakeFiles/python_module_PyKDL.dir/PyKDL/sipPyKDLpart0.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:96: CMakeFiles/python_module_PyKDL.dir/all] Error 2
make: *** [Makefile:150: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
Error making: ros-noetic-python-orocos-kdl
```

--- a/python_orocos_kdl/PyKDL/sip/dynamics.sip
+++ b/python_orocos_kdl/PyKDL/sip/dynamics.sip
@@ -93,4 +93,7 @@ public:
     int JntToCoriolis(const JntArray &q, const JntArray &q_dot, JntArray &coriolis);
     int JntToMass(const JntArray &q, JntSpaceInertiaMatrix& H);
     int JntToGravity(const JntArray &q,JntArray &gravity);
+
+private:
+    ChainDynParam& operator=(const ChainDynParam&);
 };

--- a/python_orocos_kdl/PyKDL/sip/kinfam.sip
+++ b/python_orocos_kdl/PyKDL/sip/kinfam.sip
@@ -412,6 +412,9 @@ public:
 //    Argument by reference doesn't work for container types
 //    virtual int JntToCart(const JntArray& q_in, std::vector<Frame>& p_out, int segmentNr=-1);
     virtual void updateInternalDataStructures();
+
+private:
+    ChainFkSolverPos_recursive& operator=(const ChainFkSolverPos_recursive&);
 };
 
 class ChainFkSolverVel_recursive : ChainFkSolverVel
@@ -426,6 +429,9 @@ public:
 //    Argument by reference doesn't work for container types
 //    virtual int JntToCart(const JntArrayVel& q_in, std::vector<FrameVel>& out, int segmentNr=-1 );
     virtual void updateInternalDataStructures();
+
+private:
+    ChainFkSolverVel_recursive& operator=(const ChainFkSolverVel_recursive&);
 };
 
 class ChainIkSolverPos : SolverI {
@@ -461,6 +467,9 @@ public:
 
     virtual int CartToJnt(const JntArray& q_init , const Frame& p_in ,JntArray& q_out);
     virtual void updateInternalDataStructures();
+
+private:
+    ChainIkSolverPos_NR& operator=(const ChainIkSolverPos_NR&);
 };
 
 class ChainIkSolverPos_NR_JL : ChainIkSolverPos
@@ -476,6 +485,9 @@ public:
 
     virtual int CartToJnt(const JntArray& q_init , const Frame& p_in ,JntArray& q_out);
     virtual void updateInternalDataStructures();
+
+private:
+    ChainIkSolverPos_NR_JL& operator=(const ChainIkSolverPos_NR_JL&);
 };
 
 class ChainIkSolverVel_pinv : ChainIkSolverVel
@@ -489,6 +501,9 @@ public:
 
     virtual int CartToJnt(const JntArray& q_in, const Twist& v_in, JntArray& qdot_out);
     virtual void updateInternalDataStructures();
+
+private:
+    ChainIkSolverVel_pinv& operator=(const ChainIkSolverVel_pinv&);
 };
 
 class ChainIkSolverVel_wdls : ChainIkSolverVel
@@ -598,6 +613,8 @@ public:
 
     void setLambda(const double& lambda);
 
+private:
+    ChainIkSolverVel_wdls& operator=(const ChainIkSolverVel_wdls&);
 };
 
 
@@ -612,6 +629,9 @@ public:
 
     virtual int CartToJnt(const JntArray& q_init , const Frame& p_in ,JntArray& q_out);
     virtual void updateInternalDataStructures();
+
+private:
+    ChainIkSolverPos_LMA& operator=(const ChainIkSolverPos_LMA&);
 };
 
 
@@ -638,6 +658,9 @@ public:
     const JntArray& getOptPos()const /Factory/;
 
     const double& getAlpha()const /Factory/;
+
+private:
+    ChainIkSolverVel_pinv_nso& operator=(const ChainIkSolverVel_pinv_nso&);
 };
 
 class ChainIkSolverVel_pinv_givens : ChainIkSolverVel
@@ -651,6 +674,9 @@ public:
 
     virtual int CartToJnt(const JntArray& q_in, const Twist& v_in, JntArray& qdot_out);
     virtual void updateInternalDataStructures();
+
+private:
+    ChainIkSolverVel_pinv_givens& operator=(const ChainIkSolverVel_pinv_givens&);
 };
 
 class ChainJntToJacSolver : SolverI
@@ -663,6 +689,9 @@ public:
     ChainJntToJacSolver(const Chain& chain);
     int JntToJac(const JntArray& q_in,Jacobian& jac, int seg_nr=-1);
     virtual void updateInternalDataStructures();
+
+private:
+    ChainJntToJacSolver& operator=(const ChainJntToJacSolver&);
 };
 
 class ChainJntToJacDotSolver : SolverI
@@ -690,6 +719,9 @@ public:
     void setBodyFixedRepresentation();
     void setInertialRepresentation();
     void setRepresentation(const int& representation);
+
+private:
+    ChainJntToJacDotSolver& operator=(const ChainJntToJacDotSolver&);
 };
 
 class ChainIdSolver : SolverI
@@ -712,4 +744,7 @@ public:
     ChainIdSolver_RNE(const Chain& chain,Vector grav);
     int CartToJnt(const JntArray &q, const JntArray &q_dot, const JntArray &q_dotdot, const std::vector<Wrench>& f_ext,JntArray &torques);
     virtual void updateInternalDataStructures();
+
+private:
+    ChainIdSolver_RNE& operator=(const KDL::ChainIdSolver_RNE&);
 };

